### PR TITLE
Make sure only series with chunks are sent

### PR DESCRIPTION
**Pull request by gouthamve**
Could fix https://github.com/cortexproject/cortex/issues/1354

### CODE REVIEW

**Comment by tomwilkie** [comment]
Panic in #1354 seems to be on the chunk store path, not the ingester path.

Also, might be easier to put this check in the querier side of this interface, so you can test without rolling new ingesters?

**Comment by gouthamve** [comment]
> seems to be on the chunk store path, not the ingester path.

Not really, https://github.com/cortexproject/cortex/blob/master/pkg/querier/ingester_streaming_queryable.go#L99-L103, the ingester path uses code from that file, but its still the ingester streaming path.

> Also, might be easier to put this check in the querier side of this interface, so you can test without rolling new ingesters?

Done.

File: pkg/ingester/ingester_test.go [hunk 1]
```diff
@@ -187,6 +191,32 @@ func TestIngesterAppend(t *testing.T) {
...
+		err = ing.QueryStream(req, &s)
+		require.NoError(t, err)
+
+		// Nothing should be selected.
+		require.Equal(t, 0, len(s.responses))
+	}
+
+	// Read samples back via chunk store.
+	ing.Shutdown()
+	//store.checkData(t, userIDs, testData)
```
Comment (tomwilkie) [inline pkg/ingester/ingester_test.go]: Lets not commit commented-out code.

**Comment by tomwilkie** [comment]
LGTM!

**Pull request merged**